custom_components/octopus_energy/cost_tracker/cost_tracker.py

```diff
@@ -132,22 +135,25 @@ async def async_added_to_hass(self):
   async def _async_calculate_cost(self, event: EventType[EventStateChangedData]):
     new_state = event.data["new_state"]
     old_state = event.data["old_state"]
+    accumulative = event.data["new_state"].attributes.get(CONF_STATE_CLASS, SensorStateClass.MEASUREMENT) != SensorStateClass.MEASUREMENT
     if new_state is None or new_state.state in (STATE_UNAVAILABLE, STATE_UNKNOWN) or old_state.state in (STATE_UNAVAILABLE, STATE_UNKNOWN):
       return
     
     _LOGGER.debug(f'event: {event.data}')
 
     current = now()
     rates_result: ElectricityRatesCoordinatorResult = self.coordinator.data if self.coordinator is not None and self.coordinator.data is not None else None
+    new_state_value = EnergyConverter.convert(float(new_state.state), event.data["new_state"].attributes.get(CONF_UNIT_OF_MEASUREMENT, UnitOfEnergy.KILO_WATT_HOUR), UnitOfEnergy.KILO_WATT_HOUR)
```

Comment:
Didn't know this existed, which is great.

Probably need some more validation on the unit of measurement of the target entity if we're going to rely on it. Which could technically be a breaking change :(